packages/mobile/src/pincode/authentication.test.ts

```diff
@@ -7,45 +8,92 @@ import { navigate, navigateBack } from 'src/navigator/NavigationService'
 import {
   CANCELLED_PIN_INPUT,
   DEFAULT_CACHE_ACCOUNT,
-  PinBlocklist,
   getPasswordSaga,
   getPincode,
+  getPincodeWithBiometry,
+  PinBlocklist,
+  setPincodeWithBiometry,
+  updatePin,
 } from 'src/pincode/authentication'
 import { clearPasswordCaches, getCachedPin, setCachedPin } from 'src/pincode/PasswordCache'
+import { store } from 'src/redux/store'
+import Logger from 'src/utils/Logger'
+import { getMockStoreData } from 'test/utils'
 import { mockAccount } from 'test/values'
+import { mocked } from 'ts-jest/utils'
 
-const mockPepper = { password: '0000000000000000000000000000000000000000000000000000000000000001' }
+jest.unmock('src/pincode/authentication')
+jest.mock('src/redux/store', () => ({ store: { getState: jest.fn() } }))
+const loggerErrorSpy = jest.spyOn(Logger, 'error')
+
+const mockPepper = {
+  username: 'some username',
+  password: '0000000000000000000000000000000000000000000000000000000000000001',
+  service: 'some service',
+  storage: 'some string',
+}
 const mockPin = '111555'
+const mockedKeychain = mocked(Keychain)
+const mockStore = mocked(store)
+mockStore.getState.mockImplementation(getMockStoreData)
+const mockedNavigate = navigate as jest.Mock
 
-describe(getPasswordSaga, () => {
-  const mockedNavigate = navigate as jest.Mock
+const expectPincodeEntered = () => {
+  expect(navigate).toHaveBeenCalledWith(
+    'PincodeEnter',
+    expect.objectContaining({
+      withVerification: true,
+    })
+  )
+  expect(navigateBack).toHaveBeenCalled()
+}
 
+describe(getPasswordSaga, () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockedNavigate.mockReset()
+  })
   it('Gets password', async () => {
     mockedNavigate.mockImplementationOnce((_, params) => {
       expect(params.withVerification).toBe(true)
       params.onSuccess(mockPin)
     })
 
-    const mockGetGenericPassword = Keychain.getGenericPassword as jest.Mock
-    mockGetGenericPassword.mockResolvedValue(mockPepper)
+    mockedKeychain.getGenericPassword.mockResolvedValue(mockPepper)
     const expectedPassword = mockPepper.password + mockPin
 
     await expectSaga(getPasswordSaga, mockAccount, true, false)
       .provide([[select(pincodeTypeSelector), PincodeType.CustomPin]])
       .returns(expectedPassword)
       .run()
 
-    // expect(navigate).toHaveBeenCalled()
-    // expect(navigateBack).toHaveBeenCalled()
+    expectPincodeEntered()
+  })
+  it('should throw an error for unset pincode type', async () => {
+    try {
+      await expectSaga(getPasswordSaga, mockAccount, false, false)
+        .provide([[select(pincodeTypeSelector), PincodeType.Unset]])
+        .run()
+    } catch (error) {
+      expect(error).toEqual(Error('Pin has never been set'))
+    }
```

Comment:
This will happily pass if no error is actually thrown.
Related to our discussion in https://github.com/valora-inc/wallet/pull/1816#discussion_r793482703

Reply:
✅ i've updated all these already in https://github.com/valora-inc/wallet/pull/1816